plugins/modules/system/homectl.py

```diff
@@ -0,0 +1,634 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2022, James Livulpi
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: homectl
+author:
+    - "James Livulpi (@jameslivulpi)"
+short_description: Manage user accounts with systemd-homed
+version_added: 4.4.0
+description:
+    - Manages a user's home directory managed by systemd-homed.
+options:
+    name:
+        description:
+            - The user name to create, remove, or update.
+        required: true
+        aliases: [ 'user', 'username' ]
+        type: str
+    password:
+        description:
+            - Set the user's password to this.
+            - Homed requires this value to be in cleartext on user creation and updating a user.
+            - The module takes the password and generates a password hash in SHA-512 with 10000 rounds of salt generation using crypt.
+            - See U(https://systemd.io/USER_RECORD/).
+            - This is required for I(state=present).
```

Comment:
I think the following is more precise:
```suggestion
            - This is required for I(state=present), but will B(only be used) on creation.
```
Is this correct?

Reply:
Yes, even know when modifying a user we still need the `password` argument as homed requires this to edit the user.

Reply:
Does it allow to change the password when editing the user? I.e. what happens if you provide a different password than on the previous run (for the same user)?

Reply:
I do plan to add this ( changing current users password) but as of now no.

If you provide a different password it should fail, I just found a bug. I need to add a method to check on users password is correct if its an existing user or else the homectl commands could hang waiting on input.

Reply:
Next commit will output:
```
fatal: [testhost]: FAILED! => {"changed": false, "msg": "User exists but password is incorrect!", "state": "present"}
```

If the user exists but password is incorrect, I plan to add another field at some point to change password but this will require the user to give the "known" password and the new password to change.